packages/js/src/shared-admin/components/search/search.js

```diff
@@ -0,0 +1,297 @@
+			return false;
+		}
+
+		// Lowercase and split query into words.
+		const splitQuery = split( safeToLocaleLower( trimmedQuery, userLocale ), " " );
+
+		// Filter search index by split query and store number of hits.
+		// A hit is registered if a single word from split query in found in a fields keywords.
+		const queryResults = reduce( queryableSearchIndex, ( queryResultsAcc, item ) => {
+			const hits = reduce( splitQuery, ( hitsAcc, queryWord ) => includes( item?.keywords, queryWord ) ? ++hitsAcc : hitsAcc, 0 );
+
+			// Bail if no hits found.
+			if ( hits === 0 ) {
+				return queryResultsAcc;
+			}
+
+			return [
+				...queryResultsAcc,
+				{
+					...item,
+					// Store hits for later sorting.
+					hits,
+				},
+			];
+		}, [] );
+
+		// Sort query results by number of hits on field, highest number of hits first.
+		const sortedQueryResults = queryResults.sort( ( a, b ) => b.hits - a.hits );
+
+		// Group query results by route.
+		const groupedQueryResults = groupBy( sortedQueryResults, "route" );
+
+		// Sort route groups by max hits of single field, highest number of max hits first.
+		// Ie. if group A contains a field with 2 hits and group B contains 2 fields with 1 hit, group A will be sorted first.
+		const sortedGroupedQueryResults = values( groupedQueryResults ).sort( ( a, b ) => {
+			const aMaxHits = reduce( a, ( maxHitsAcc, queryResult ) => max( [ maxHitsAcc, queryResult.hits ] ), 0 );
+			const bMaxHits = reduce( b, ( maxHitsAcc, queryResult ) => max( [ maxHitsAcc, queryResult.hits ] ), 0 );
+			return bMaxHits - aMaxHits;
+		} );
+
+		if ( isEmpty( queryResults ) ) {
+			setA11yMessage( __( "No results found", "wordpress-seo" ) );
+		} else {
+			setA11yMessage( sprintf(
+				/* translators: %d expands to the number of results found. */
+				_n(
+					"%d result found, use up and down arrow keys to navigate",
+					"%d results found, use up and down arrow keys to navigate",
+					queryResults.length,
+					"wordpress-seo"
+				),
+				queryResults.length
+			) );
+		}
+
+		setResults( sortedGroupedQueryResults );
+	}, 100 ), [ queryableSearchIndex, userLocale, setA11yMessage ] );
+
+	const handleQueryChange = useCallback( event => {
+		setQuery( event.target.value );
+		debouncedSearch( event.target.value );
+	}, [ setQuery, debouncedSearch ] );
+
+	const handleOptionActiveState = useCallback( ( { active } ) => classNames(
+		"yst-group yst-block yst-no-underline yst-text-sm  yst-select-none yst-py-3 yst-px-4 hover:yst-bg-primary-600 hover:yst-text-white focus:yst-bg-primary-600 focus:yst-text-white",
+		active ? "yst-text-white yst-bg-primary-600" : "yst-text-slate-800"
+	), [] );
+
+	return <>
+		<Button
+			id={ buttonId }
+			type="button"
+			variant="secondary"
+			className="yst-w-full yst-flex yst-items-center yst-justify-start yst-font-normal yst-text-sm yst-leading-6 yst-text-slate-500 yst-rounded-md yst-border-slate-300 yst-shadow-sm yst-py-1.5 yst-pl-2 yst-pr-3"
```

Comment:
I believe some of the classes here are already part of the `secondary` variant and the button component.